AUTOMATAFORMAT.md

```diff
@@ -0,0 +1,111 @@
+
+# The file format for automata
+* Based partially on Ondrej Lengal's [`.vtf' format](https://discord.com/channels/@me/864885374375821312/980792642927460372), a format of Pavol Vargovcik, and a confused disucssion.
+
+## Top-level file structure
```

Comment:
I meant comments up to the end of line (excluding the trailing `\`, which concatenates lines) to be able to do:
```
qHello \
  (aWave & aSmile & qHi)   %# This handles proper greeting \
  & ((aKnown & qTalk) | qIntroduce)  %# Introduce yourself to unknown people
```

I agree that multiline comment blocks are useless.
Maybe single line comment blocks are more useful but not so important:

```
qIntroduce (aSayYourName & aRememberHerName %# This is often forgotten by humans #% & qTalk)
```

Reply:
So every line starting with `#` or `//` is comment. Nothing else.

Reply:
why is it so important to make it visually different? it is already quite visually different because it is a human readable text. I propose the following solution:
- comment is a special instance of attribute (we've already agreed on this for inline comments).
- attributes start with the letter `t`
- a line can be a keyword definition (starting with `%`) or an identifier definition (states `q`, nodes `n` and why not attributes `t`)

This way, we don't add anything new to the language and we can write comment lines as:
```
t"The following state is good for nothing."
qUseless false
```

Reply:
It's not about visuality but simplicity. We agreed that we do not need inline comments and for sake of parsing simplicity we would like to use `#` or `//` and filter everything on the line beginning with this.

Reply:
I like // or # since it looks normal, easy to explain, and it is immediately clear that it is a line to ignore.  t"bla" is something multipurpose and one needs to check what the bla actually is to identify it as a line to ignore. Also exaplining needs some nontrivial sentences - the concept that comments are actually attributes, where attributes are ...  right now I don't even remember ...

Reply:
A situation when t"bla bla" is pretty bad is when commenting out a bunch of lines with transitions. You have to write t" at the beginning of each line and another " at the end of each line.

Reply:
`#` vyhral